### ISSUE TITLE
User substitution variables aren't applying

### ISSUE THREAD

**Issue by aslattery** [googleapis/nodejs-cloudbuild#117]
I'm experiencing an issue where attempting to provide user substitution variables via `substitutions` property of a [`RepoSource` object](https://googleapis.dev/nodejs/cloudbuild/latest/google.devtools.cloudbuild.v1.RepoSource.html) passed via `runBuildTrigger` is not reflected in my build steps.

#### Environment details

  - OS: GCF, HTTP Trigger
  - Node.js version: 10
  - npm version: Unknown
  - `@google-cloud/cloudbuild` version: 1.5.0

#### Steps to reproduce

  1. Have a Google Cloud Build trigger with a few user substitution variables added
  2. Declare an object of `substitutions` following the naming convention (e.g. `_MY_SUB_VAR`)
  3. Review your GCB Build Log, both in a bash log and execution details tab, default values under "User substitutions" are still present; my provided substitutions are ignored

Of note, when breaking the naming convention (omitting the `_` prefix), the build does fail due to GCB validation; so the object _is_ being sent in the request, and appears under "Built-in substitutions" in the build log; when using the proper naming convention, they do not map over the defaults.

![Annotation 2020-05-12 195521](https://user-images.githubusercontent.com/5545673/81766208-88d0bb80-948a-11ea-957c-2f815e29eaff.png)

#### Redacted Code Snippet of `runBuildTrigger` invocation

```javascript
gcb.runBuildTrigger(
    {
        projectId: "...",
        triggerId: "...",
        source: {
            projectId: "...",
            branchName: targetBranchOrTag,
            substitutions: {
                _APP_BASE_SERVICE_URI: `https://${targetBucket[0].name}`,
                _RS_BASE_SERVICE_URI: `https://.../${targetBranchOrTag}`,
                _TARGET_BUCKET: targetBucket[0].name,
            },
        },
    },
...
```

Any guidance or assistance in resolving this matter is appreciated.

**Comment by aslattery** [comment]
Since filing this issue originally, I have also attempted to workaround this by using the [`googleapis`](https://googleapis.dev/nodejs/googleapis/latest/cloudbuild/index.html) Node client, and face the same issue: substitutions provided in the `RepoSource` object aren't being mapped over defaults at the time of execution.

Doing a bit more trial and error on my part this afternoon, I used the "Try this API" tooling [on the cloud build REST API reference](https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.triggers/run), and effectively mirror the `projects.triggers.run` request, it does succeed  in mapping my provided substitutions at execution!

@bcoe, after seeing your reply on #119 ([ref](https://github.com/googleapis/nodejs-cloudbuild/issues/119#issuecomment-638541221)), should I also report this issue upstream to the [Cloud Build issue tracker](https://issuetracker.google.com/issues/new?component=190802&template=1162743)?

Apologies for a newbie question, but after years of implementing various Google client API libraries, I've apparently been very fortunate to not run into an issue I couldn't find a resolution for 😅

**Comment by bcoe** [comment]
@aslattery substitutions are working as expected for me, I created this minimal configuration for testing:

```
steps:
- name: node:13
  entrypoint: npm
  args: ['install', "${_MY_TEST_SUBSTITUTION}"]
```

I then used the following command with the API:

```js
  // Starts a build against the branch provided.
  const [resp] = await cb.runBuildTrigger({
    projectId,
    triggerId,
    source: {
      projectId,
      dir: './',
      branchName,
      substitutions: {
        _MY_TEST_SUBSTITUTION: "yargs",
        _RS_BASE_SERVICE_URI: "https://another-variable-that-we-substitute"
      }
    },
  });
```

Is there a chance you perhaps have `RS_BASE_SERVICE_URI` is defined elsewhere, perhaps in as a stanza in the `cloudbuild.yml` you're running programmatically?

**Comment by aslattery** [comment]
Appreciate the response, @bcoe 

I just re-enabled the trigger on my end to verify I was still experiencing the issue; substitution values are working as expected now. No changes to our `cloudbuild.yml`, I did also double check that I had rolled back the Cloud Function code to what it was when I had originally reported the issue (reverting from using the `googleapis` client).

In the context of this issue, it's clearly **not** this library, so I'll go ahead and close this, and hope it continues working upstream. Appreciate your efforts in working through this with me 👍
